File: dex-mixin-api/src/main/java/com/huanli233/dex_mixin/api/utils/SmaliUtil.java
package com.huanli233.dex_mixin.api.utils;

import org.jetbrains.annotations.NotNull;

public class SmaliUtil {

    @NotNull
    public static String javaClassToSmaliType(@NotNull String className) {
        return "L" + className.replace(".", "/") + ";";
    }

    @NotNull
    public static String javaClassToSmaliType(@NotNull Class<?> clazz) {
        return javaClassToSmaliType(clazz.getName());
    }

    @NotNull
    public static String smaliTypeToJava(@NotNull String smaliType) {
        if (smaliType.length() < 2) return smaliType;
        return smaliType.substring(1, smaliType.length() - 1).replace("/", ".");
    }

}

File: dex-mixin-api/src/main/java/com/huanli233/dex_mixin/api/utils/SmaliUtil.kt
package com.huanli233.dex_mixin.api.utils

import kotlin.reflect.KClass

fun String.classToSmaliType() =
    SmaliUtil.javaClassToSmaliType(this)

val Class<*>.smaliType: String
    get() = SmaliUtil.javaClassToSmaliType(this)

val KClass<*>.smaliType: String
    get() = java.smaliType